apps/src/applab/applab.js

```diff
@@ -1480,6 +1490,16 @@ Applab.penColor = function (opts) {
   }
 };
 
+Applab.speed = function (opts) {
+  if (opts.percent >= 0 && opts.percent <= 100) {
```

Comment:
if these conditions are not met, it might be better to default to 0 or 1 rather than doing nothing.